content/courses/advanced-ada/chapters/access_types.rst

```diff
@@ -1577,17 +1577,732 @@ type) and display its information by calling the :ada:`Show` procedure.
+          A2 := I_Var_2'Access;
+          --   PROBLEM: A2 has the same lifetime as I_Var_2,
+          --            but Integer_Access type has a longer
+          --            lifetime.
+
+          Put_Line ("A2.all: " & Integer'Image (A2.all));
+       end Inner_Block;
+
+       Put_Line ("Inner_Block has ended!");
+       Put_Line ("A1.all: " & Integer'Image (A1.all));
+
+    end Show_Dangling_Reference;
+
+Here, we declare the access objects :ada:`A1` and :ada:`A2` of
+:ada:`Integer_Access` type, and the :ada:`I_Var_1` and :ada:`I_Var_2` objects.
+Moreover, :ada:`A1` and :ada:`I_Var_1` are declared in the scope of the
+:ada:`Show_Dangling_Reference` procedure, while :ada:`A2` and :ada:`I_Var_2`
+are declared in the :ada:`Inner_Block`.
+
+When we try to compile this code, we get two compilation errors due to
+violation of accessibility rules. Let's now discuss these accessibility rules
+in terms of lifetime, and see which problems they are preventing in each case.
+
+1. In the :ada:`A1 := I_Var_2'Access` assignment, the main problem is that
+   :ada:`A1` has a longer lifetime than :ada:`I_Var_2`. After the
+   :ada:`Inner_Block` finishes |mdash| when :ada:`I_Var_2` gets out of scope
+   and its lifetime has ended |mdash|, :ada:`A1` would still be pointing to an
+   object that does not longer exist.
+
+2. In the :ada:`A2 := I_Var_2'Access` assignment, however, both :ada:`A2` and
+   :ada:`I_Var_2` have the same lifetime. In that sense, the assignment may
+   actually look pretty much OK.
+
+   - However, as mentioned in the previous section, Ada also cares about the
+     lifetime of access types. In fact, since the :ada:`Integer_Access` type is
+     declared outside of the :ada:`Inner_Block`, it has a longer lifetime than
+     :ada:`A2` and :ada:`I_Var_2`.
+
+   - To be more precise, the accessibility rules detect that :ada:`A2` is an
+     access object of a type that has a longer lifetime than :ada:`I_Var_2`.
+
+At first glance, this last accessibility rule may seem too strict, as both
+:ada:`A2` and :ada:`I_Var_2` have the same lifetime |mdash| so nothing bad
+could occur when dereferencing :ada:`A2`. However, consider the following
+change to the code:
+
+.. code-block:: ada
+
+          A2 := I_Var_2'Access;
+
+          A1 := A2;
+          --    PROBLEM: A1 will still be referring to
+          --             I_Var_2 after the Inner_Block,
+          --             i.e. when the lifetime of
+          --             I_Var_2 has ended!
+
+Here, we're introducing the :ada:`A1 := A2` assignment. The problem with this
+is that :ada:`I_Var_2`\ 's lifetime ends when the :ada:`Inner_Block` finishes,
+but :ada:`A1` would continue to refer to an :ada:`I_Var_2` object that doesn't
+exist anymore |mdash| thereby creating a dangling reference.
+
+Even though we're actually not assigning :ada:`A2` to :ada:`A1` in the original
+code, we could have done it. The accessibility rules prevent that such an error
+is ever introduced to the program.
```

Comment:
Same issue here. Suggest: "The accessibility rules ensure that such an error is never introduced into the program."